runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java

```diff
@@ -471,13 +479,29 @@ private void applyHiscoreResult(HiscoreResult result)
 			}
 			else if ((s = result.getSkill(skill)) != null)
 			{
+				label.setForeground(Color.WHITE);
 				final long exp = s.getExperience();
 				final boolean isSkill = skill.getType() == HiscoreSkillType.SKILL;
-				int level = -1;
-				if (config.virtualLevels() && isSkill && exp > -1L)
+				long level = -1;
+				if (config.maxedSkillsStyle() == MaxedSkillsStyle.VIRTUAL_LEVELS && isSkill && exp > -1L)
```

Comment:
I believe all of the `exp > -1L` can be replaced with a bool like `hasXp`?

Reply:
I simplified a lot of the code here, it seems correct to me.